js2/webhosting/lesson.md

```diff
@@ -0,0 +1,133 @@
+Obě služby jsou hodně podobné, fungují na podobných principech.
+My si ukážeme práci s Netlify, ale velmi podobný postup použijete i s Vercelem, jenom obrazovky vypadají trochu jinak.
+
+Firma, která provozuje Vercel, je zároveň autorem Reactového frameworku [Next.js](https://nextjs.org).
+Takže pokud máte projekt založený na Next.js, je Vercel jasná volba.
+Ne že by Next.js nešel provozovat jinde, ale s Vercelem je to nejsnazší.
+
+A jak je to s cenou?
+Bude pro vás někdo jen tak platit provoz serveru, abyste na něm mohly mít zdarma své stránky?
+Ano, bude 😀
+Netlify i Vercel jsou placené služby, ale obě mají základní variantu zdarma, která je dostatečná pro soukromé nebo hobby projekty.
+
+## Deploy nebo-li zveřejnění projektu
+Obě dvě služby – Netlify i Vercel – fungují tak, že propojjíte službu se svým GitHub účtem, řeknete, které repository chcete jako web zveřejnit, doplníte pár informací o tom, jak se s vaším projektem má zacházet – a Netlify nebo Vercel si stáhne zdrojové kódy, *vybuildí* aplikaci, tj. převede ji do tvaru, v jakém se dá zveřejnit na webu, a nakonec ji nahraje na nějaký server a web zveřejní.
+
+Jak už bylo řečeno v lekci o WebPacku, dnešní weby se prohlížeči neposkytují v tom tvaru, jak je napsaný zdrojový kód.
+Námi napsaný zdrojový kód prochází tzv. buildem (sestavením), kdy se vezmou HTML soubory, CSS, skripty, a zmenší se a zabalí do balíčků, aby se s nimi prohlížeči lépe pracovalo.
+Podobně se upraví třeb ai obrázky – mohou se zkomprimovat, aby zabíraly menší objem a rychleji se stáhly, případně se i přizpůsobí pro různě velká zařízení.
+Pro nás teď ale není podstatné, co vše se při buildu dělá – to zařídil někdo, kdo pro nás napsal `webpack.config.js` nebo jiný předpis buildu.
+Pro nás je teď důležité, že celý proces buildu spustíme jedním příkazem:
+
+```shell
+npm run build
+```
+
+Příkaz sestaví web do výsledné podoby, v jaké má být vystaven na webovém serveru, a jak si z něj bude prohlížeč stahovat jednotlivé soubory.
+Výsledek umístí do adresáře `dist`, který vznikl ve vašem projektu.
+V jiných projektech (s jinou konfigurací WebPacku) se cílový adresář může jmenovat třeba `build`, v našich projektech vytvořených pomocí `npm init kodim-app` se vždy bude jmenovat `dist`.
+Mimochodem, tento adresář nepatří do gitu, nebude na GitHubu (o to se zase stará konfigurace v souboru `.gitignore`).
+
+Kdybyste přeci jen použily PHP hosting, nyní nahrajete obsah adresáře `dist` na server ručně (nejspíš pomocí FTP nebo SFTP).
+Cloud hostingy se ale o tohle postarají automaticky – naklonují si repozitory, spustí build a výsledný adresář zveřejní na svém serveru.
+Jenom jim musíte pomoci s tím, kterým příkazem se build spouští (v našem případě `npm run build`) a kde bude výsledný web (u nás `dist`).
+
+To je vše, Netlify i Vercel vám po úspěšném buildu napíšou, na jakou doménu váš web nasadili.
+Bude to doména ve tvaru `*.netlify.app` nebo `*.vercel.app`, takže třeba `jumping-squirrel-7586.netlify.app`.
```

Comment:
Trošku láká na tu doménu něco nasadit. :D Nebo `jumping-squirrel-7586` máš pod kontrolou, i když ukazuje 404?

Reply:
Ne, tu doménu obsazenou nemám :-)

Reply:
Je to asi jedno, ale pobavilo mne to. :D